web/modules/custom/portland_legacy_redirects/src/Plugin/Validation/Constraint/RelativePathConstraintValidator.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace Drupal\portland_legacy_redirects\Plugin\Validation\Constraint;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Drupal\Core\Field\FieldItemListInterface;
+use Symfony\Component\Validator\Violation\ConstraintViolationBuilderInterface;
+
+/**
+ * Validates the PreventAnon constraint.
+ */
+class RelativePathConstraintValidator extends ConstraintValidator {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validate($field, Constraint $constraint) {
+
+    // is it single value or FieldItemList?
+    if ($field instanceof FieldItemListInterface && $field->getName() == "field_redirects") {
+      foreach($field as $delta => $value) {
+        $path = $value->value;
+
+        $entity = $field->getEntity();
+        $original = $entity->original;
+
+        // test if path is bare (no protocol)
+        if (!$this->validateRelative($path)) {
+          // remove protocol from path for use as a suggestion in the validation message
+          $relative_path = parse_url($path, PHP_URL_PATH);
+          $message = $constraint->not_relative . " Try using \"$relative_path\" instead.";
+          $this->setViolation($message, $delta);
```

Comment:
The links in the validation error messages that are supposed to jump the user down to the offending field don't work. It looks like "-value" needs to be appended to the $delta fieldname value used in setViolation(). (It looks like this is also broken on PGOV, but we might as well fix it here since it's an easy fix.)

Reply:
Append -value here didn't change the URL hash. I'd have to dig deeper. Please let me know if you are aware of a solution.

Reply:
Hmm. yeah, I took a quick look too and it seems like maybe it's a bug in the underlying Symfony component.

Reply:
Thanks for looking into this. The other way to fix this in Drupal is to set the ID attribute of the field to "edit-field-redirects-0" so the links in the error message would be valid. But adding that would touch every field in the form. Not sure if that would cause issues.